### PR TITLE
Task#15 add informative error messages

### PR DESCRIPTION
**Pull request by vaambival**
Added error messages.
Added negative test cases.

### CODE REVIEW

File: src/test/java/ru/curs/xylophone/TestReader.java [hunk 1]
```diff
@@ -196,6 +201,53 @@ public void testSAXReader3() throws XML2SpreadSheetError {
 		reader.process();
 		assertEquals("Q{TCQ{CCQ{CC}C}TQ{CQh{CCC}}TQ{}}F", w.getLog().toString());
 	}
+
+	@Test
+	public void testParsingDescriptorWithElementInsideElementShouldFail() throws XML2SpreadSheetError {
+		descrStream = TestReader.class
+				.getResourceAsStream("testdescriptor_with_element_inside_element.xml");
```
Comment (ioanngolovko) [inline src/test/java/ru/curs/xylophone/TestReader.java]: `test_descriptor_...` would be better, if you use underscore.

File: src/test/java/ru/curs/xylophone/TestReader.java [hunk 2]
```diff
@@ -196,6 +201,53 @@ public void testSAXReader3() throws XML2SpreadSheetError {
...
+		DummyWriter w = new DummyWriter();
+		// При создании reader (при парсинге xml) падает исключение о некорректной последовательности тегов.
+		XMLDataReader.createReader(dataStream,
+				descrStream, false, w);
+	}
+
+	@Test
+	public void testParsingDescriptorWithOutputInsideIterationShouldFail() throws XML2SpreadSheetError {
+		descrStream = TestReader.class
+				.getResourceAsStream("testdescriptor_with_output_inside_iteration.xml");
```
Comment (ioanngolovko) [inline src/test/java/ru/curs/xylophone/TestReader.java]: test_descriptor_... would be better, if you use underscore.

File: src/test/java/ru/curs/xylophone/TestReader.java [hunk 3]
```diff
@@ -196,6 +201,53 @@ public void testSAXReader3() throws XML2SpreadSheetError {
...
+		DummyWriter w = new DummyWriter();
+		// При создании reader (при парсинге xml) падает исключение о некорректной последовательности тегов.
+		XMLDataReader.createReader(dataStream,
+				descrStream, false, w);
+	}
+
+	@Test
+	public void testParsingDescriptorWithIterationInsideIterationShouldFail() throws XML2SpreadSheetError {
+		descrStream = TestReader.class
+				.getResourceAsStream("testdescriptor_with_iteration_inside_iteration.xml");
```
Comment (ioanngolovko) [inline src/test/java/ru/curs/xylophone/TestReader.java]: test_descriptor_... would be better, if you use underscore.

File: src/test/java/ru/curs/xylophone/TestReader.java [hunk 4]
```diff
@@ -196,6 +201,53 @@ public void testSAXReader3() throws XML2SpreadSheetError {
...
+		dataStream = TestReader.class.getResourceAsStream("testdata.xml");
+
+		expectedException.expect(XML2SpreadSheetError.class);
+		expectedException.expectMessage("Tag <iteration> is not allowed inside <iteration>. " +
+				"Error inside element with name titlepage.");
+
+		DummyWriter w = new DummyWriter();
+		// При создании reader (при парсинге xml) падает исключение о некорректной последовательности тегов.
```
Comment (ioanngolovko) [inline src/test/java/ru/curs/xylophone/TestReader.java]: Xylophone is opensource project. Please, use english language in comments and javadocs.

File: src/test/resources/ru/curs/xylophone/test_descriptor_with_element_inside_element.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<element name="report">
+    <!-- Вывод единственного титульного листа, с иерархией -->
```
Comment (ioanngolovko) [inline src/test/resources/ru/curs/xylophone/test_descriptor_with_element_inside_element.xml]: =)

**Review by ioanngolovko — APPROVED** [review]

File: src/test/resources/ru/curs/xylophone/test_descriptor_with_element_inside_element.xml (hunk 5 above)
Comment (vaambival) [reply src/test/resources/ru/curs/xylophone/test_descriptor_with_element_inside_element.xml]: Fixed remark=)

**Pull request merged**